controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -136,100 +145,113 @@ func (r *RateLimitPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl
 
 func (r *RateLimitPolicyReconciler) reconcileWithVirtualService(ctx context.Context, vs *istio.VirtualService, rlp *apimv1alpha1.RateLimitPolicy) error {
 	logger := r.Logger()
+	rlpKey := client.ObjectKeyFromObject(rlp)
 
 	_, ok := vs.Annotations[VSAnnotationRateLimitPolicy]
```

Comment:
why is the VS being annotated? What's the purpose?

Reply:
To create a reconcile request if it changes.

Reply:
But it is not implemented, is it?

If it is not, I would leave for another PR.

Anyway, if you want to reconcile EnvoyFilters if the VS changes, the annotation is not strictly needed. There is another way. You just need to add a watch of the VS kind with mapper that gets the list of RLF objects filtering by the `spec.networkingRef`.

Reply:
I thought about doing it that way but it felt like a little extra work that can be solved easily with an annotation but if we find it in the future interacting with users need in a negative way, can follow this route.